apps/trackers/jira/query.py

```diff
@@ -585,6 +584,105 @@ def generate(self):
         self.generate_embargo_status()
         self.generate_special_handling()
 
+    @cached_property
+    def most_important_affect(self):
+        """
+        Selects the most important affect to use for displaying
+        CVE/CVSS/CWE/Source in Vulnerability issuetype Jira tracker
+        fields, which is relevant for multi-flaw trackers where
+        the tracker can have multiple sets of CVE/CVSS/CWE/source
+        but the tracker should display only one such set.
+        Quote from 2024-09-27 in OSIDB-3348:
+          "For multiflaw trackers, select CVE with highest Impact value,
+           if there are multiple such CVE, select the one where flaw has
+           the oldest created_date/time value. Use CVE/cvss/cwe/source
+           from that CVE."
+        Note: The writer assumed a flaw has only one authoritative CVE
+              object (the one issued by RH) and Flaw == CVE in this context.
+        Note: Assuming validations and administrative procedures ensure
+              that the impact model fields are in accordance with the
+              cvss_scores fields.
+        """
+        affects_impacts = sorted(
+            [
+                (affect, affect.aggregated_impact)
+                for affect in self.tracker.affects.all()
+            ],
+            key=lambda pair: pair[1],
+        )
+        greatest_impact_affects_impacts = [
+            (affect, impact)
+            for affect, impact in affects_impacts
+            if impact == affects_impacts[-1][1]
+        ]
+        flaw_created_dt_affects = sorted(
+            [
+                (affect.flaw.created_dt, affect)
+                for affect, impact in greatest_impact_affects_impacts
+            ],
+            key=lambda pair: pair[0],
+        )
+        most_important_affect = flaw_created_dt_affects[0][1]
```

Comment:
Maybe directly using the queryset's `order_by` would be shorter (and probably more performant?) than using a list comprehension and sorting in Python?

Reply:
`aggregated_impact` is not a DB field.

```
django.core.exceptions.FieldError: Cannot resolve keyword 'aggregated_impact' into field. Choices are: acl_read, acl_write, affectedness, alerts, created_dt, cvss_scores, embargoed, events, exploitreview, flaw, flaw_id, impact, meta_attr, ps_component, ps_module, resolution, trackers, updated_dt, uuid
```

So I do all sorting in python to at least keep the principles similar inside the single function.

The amount of objects iterated this way in `greatest_impact_affects_impacts` and `flaw_created_dt_affects` is IMO small enough not to be a performance issue.